materials/databases_intro.md

```diff
@@ -5,81 +5,104 @@ title: Introduction to Databases
 
-**Open SQLite w/Portal DB**
+> Open and preview SQLite with `portal_mammals.sqlite`
 
-## Key features of database management systems
+### Key features of database management systems
 
 * Data is separate from manipulations of the data
 * Tables - store the data
 * Queries - store questions about the data
     * If we update the data, the query asks the same question of the new data
 
-## Relational databases
-
-**Demo in SQLite**
+### Relational databases
 
 * Data is stored in tables
     * One table per type of data
-	* Tables can be linked together to combine information
+    * Tables can be linked together to combine information
 * Each row contains a single record
     * A single observation or data point
 * Each column contains a single attribute
-    * A single type of information
+    * A single field or type of information
 
-## Types
+### Data Types
 
 * Fields in databases have types that define the kind of data they contain
 * Each field/column can only have one type
 * We have to define the types in advance
 * Types include
     * Integer
-	* Text
-	* Decimal/Double
-* Types are highly configurable because space is limiting
+    * Text
+    * Decimal/Double/Real/Float
+* Types are highly configurable for when space is limiting
     * Maximum value of integers
-	* Maximum length of text
-	* How many values before and after the decimal place
+    * Maximum length of text
+    * How many values before and after the decimal place
 
-## Primary keys
+### Primary keys
 
 * Every table "needs" a column (or set of columns) that is unique across
   records/row
-* This is called the primary key
-* The easiest way to do this is to use an Integer that increments every time a
+* This is called the 'Primary Key'
+* The easiest way to do this is to use an 'Integer' that increments every time a
   new record is added
 * Many databases that you import will already have a field like this
 
+### Database Control
+
+There are many ways to add and modify data in a database management system.
+
+1\. Enter the data into another program (e.g., a spreadsheet) and import it to SQLite
+
+> Do [Exercise 1 - Data entry validation in Excel]({{ site.baseurl }}/exercises/Qaqc-data-entry-validation-in-excel-SQL/). 
+
+2\. Get the data from somewhere and import it ( *though, data isn't always ready 
+to import into a database* )
+
+> Do [Exercise 2 - Importing Data]({{ site.baseurl }}/exercises/Database-control-importing-data-SQL).
+
+3\. Enter the data directly.
+
+> Demo [Creating Tables]({{ site.baseurl }}/materials/sql-creating-tables).
+>
+> Do [Exercise 3 - Creating Tables]({{ site.baseurl }}/exercises/Database-control-creating-tables-SQL/).
+
+> Demo [Updating Tables]({{ site.baseurl }}/materials/sql-updating-tables).
+>  
+> Do [Exercise 4 - Adding Records]({{ site.baseurl }}/exercises/Database-control-adding-records-SQL/) and [Exercise 5 - Updating Records]({{ site.baseurl }}/exercises/Database-control-updating-records-SQL/).
+
```

Comment:
In general I think this section should start with a quick overview of the options, not offset with exercises, and then provide brief introductions to how to do the thing in the bullet point before having the students work on the exercise. It looks like some of the introductory material is in separate files, but it seems like it would be most useful to integrate it into the lecture directly. In cases where that material doesn't exist it will need to be added, either now or by creating new issues and taking care of it later.